File: Behavioral/Chain of responsibility/C#/ChainOfResponsibilityExample/ChainOfResponsibilityExample/SecondHandler.cs
namespace ChainOfResponsibilityExample;

class SecondHandler : BaseHandler
{
	public override void Handle(string request)
	{
		foreach (char ch in request)
		{
			if (Char.IsDigit(ch))
			{
				Console.WriteLine("The second handler processed: " + request);
				return;
			}
		}

		base.Handle(request);
	}
}

File: Behavioral/Chain of responsibility/Delphi/SecondHandler.pas
unit SecondHandler;

interface

uses
  BaseHandler;

type
  TSecondHandler = class(TBaseHandler)
    procedure Handle(request: string); override;
  end;

implementation

uses
  SysUtils;

procedure TSecondHandler.Handle(request: string);
var
  i: Integer;
begin
  for i := 1 to Length(request) do
    if CharInSet(request[i], ['0'..'9']) then begin
      Writeln('The second handler processed:', request);
      Exit;
    end;
  inherited Handle(request);
end;

end.
